scripts/Muon/GUI/Common/plotting_widget/plotting_widget_model.py

```diff
@@ -7,202 +7,238 @@
-        self.plotted_workspaces = []
-        self.plotted_workspaces_inverse_binning = {}
-        self.plotted_fit_workspaces = []
-
-    def force_redraw(self):
-        if not self.plot_figure:
+        try:
+            workspace = AnalysisDataService.Instance().retrieve(workspace_name)
+        except RuntimeError:
             return
 
-        self.plot_figure.canvas.draw()
-
-    def autoscale_y_to_data_in_view(self):
-        axis = self.plot_figure.gca()
-        xlim = axis.get_xlim()
-        ylim = np.inf, -np.inf
-        for line in axis.lines:
-            x, y = line.get_data()
-            start, stop = np.searchsorted(x, xlim)
-            y_within_range = y[max(start - 1, 0):(stop + 1)]
-            ylim = min(ylim[0], np.nanmin(y_within_range)), max(ylim[1], np.nanmax(y_within_range))
+        axis.replace_workspace_artists(workspace)
 
-        new_bottom = ylim[0] * 1.3 if ylim[0] < 0.0 else ylim[0] * 0.7
-        new_top = ylim[1] * 1.3 if ylim[1] > 0.0 else ylim[1] * 0.7
-
-        axis.set_ylim(bottom=new_bottom, top=new_top)
+    def clear_plot_model(self, axes):
+        self._remove_all_data_workspaces_from_plot(axes)
+        self.plotted_workspaces = []
+        self.plotted_workspaces_inverse_binning = {}
+        self.plotted_fit_workspaces = []
 
-    def _remove_all_data_workspaces_from_plot(self):
+    def set_x_lim(self, domain, axes):
+        if domain == "Time":
+            ymin, ymax = self._get_autoscale_y_limits(axes, 0, 15)
+            plt.setp(axes, xlim=[0, 15.0], ylim=[ymin, ymax])
+        if domain == "Frequency":
+            ymin, ymax = self._get_autoscale_y_limits(axes, 0, 50)
+            plt.setp(axes, xlim=[0, 50.0], ylim=[ymin, ymax])
+
+    def _get_autoscale_y_limits(self, axes, xmin, xmax):
+        new_bottom = 1e9
+        new_top = -1e9
+        for i in range(self.number_of_axes):
+            axis = axes[i]
+            axis.set_xlim(left=xmin, right=xmax)
+            xlim = axis.get_xlim()
+            ylim = np.inf, -np.inf
+            for line in axis.lines:
+                x, y = line.get_data()
+                start, stop = np.searchsorted(x, xlim)
+                y_within_range = y[max(start - 1, 0):(stop + 1)]
+                ylim = min(ylim[0], np.nanmin(y_within_range)), max(ylim[1], np.nanmax(y_within_range))
+
+            new_bottom_i = ylim[0] * 1.3 if ylim[0] < 0.0 else ylim[0] * 0.7
+            new_top_i = ylim[1] * 1.3 if ylim[1] > 0.0 else ylim[1] * 0.7
+            if new_bottom_i < new_bottom:
+                new_bottom = new_bottom_i
+            if new_top_i > new_top:
+                new_top = new_top_i
+
+        return new_bottom, new_top
+
+    def _remove_all_data_workspaces_from_plot(self, axes):
         workspaces_to_remove = self.plotted_workspaces
-        for workspace in workspaces_to_remove:
-            self.remove_workpace_from_plot(workspace)
+        for i, workspace in enumerate(workspaces_to_remove):
```

Comment:
Why do we need to enumerate here. As far as I can tell the i isn't used for anything?

Reply:
Good spot, I used to use the variable, but it is now no longer needed.